### PR TITLE
Bugfix #47 ktps 1320 train feature names

### LINKED ISSUE #47: [BUG] Error for train_model
**Describe the bug**
On Alliander-PRD the train_model task raises an exception for specific models.
Looks like bad error-handling when new model has different features compared to old model.
See screenshot

**To Reproduce**
1. Train a model
2. specify new feature-set hyperparameter
3. retrain model -> exception

**Expected behavior**
Not an exception

**Screenshots**
image

### PR DESCRIPTION
**Pull request by FrankKr**
Fixes #47 and KTPS-1320
I added an xgboost-specific check when comparing a new model to the previous model.
It checks if the feature names are the same. If not, the new model is considered better. 
I added a TODO in the code suggesting an improvement when this section of the code is restructured.
Moreover, I fixed the `test_train_more.py` unittests which had been disabled at the start of the open-sourcing. 

For review: Please comment if docstrings are unclear

### CODE REVIEW

File: openstf/model/trainer/xgboost/xgboost.py [hunk 1]
```diff
@@ -176,35 +197,27 @@ def better_than_old_model(self, test_data, penalty_factor=1.2):
                 )
             except Exception as e:
                 prediction_old_model = np.nan
```
Comment (JanMaartenvanDoorn) [inline openstf/model/trainer/xgboost/xgboost.py:199]: This variable seems not to be used further in the code

File: openstf/model/trainer/xgboost/xgboost.py [hunk 2]
```diff
@@ -104,6 +109,7 @@ def train(
             validation_data (pandas.DataFrame): The validation data.
             callbacks (list of callable): List of callback functions that can be
                 called at the end of each training iteration
+            num_boost_round (int): can be large since we use early stopping.
```
Comment (JanMaartenvanDoorn) [inline openstf/model/trainer/xgboost/xgboost.py:112]: early stopping rounds seem to be missing in the docstring

**Review by JanMaartenvanDoorn — APPROVED** [review]
Looks good! Two minor things to improve.

File: openstf/model/trainer/xgboost/xgboost.py (hunk 2 above)
Comment (FrankKr) [reply openstf/model/trainer/xgboost/xgboost.py:114]: I'll include it!

File: openstf/model/trainer/xgboost/xgboost.py (hunk 1 above)
Comment (FrankKr) [reply openstf/model/trainer/xgboost/xgboost.py]: thanks, I'll remove it!

**Pull request merged**